lib/qmi/codec/network_access.ex

```diff
@@ -319,5 +349,19 @@ defmodule QMI.Codec.NetworkAccess do
   defp serving_system_network(0x01), do: :network_3gpp2
   defp serving_system_network(0x02), do: :network_3gpp
 
-  defp calc_tz_offset(tz_offset), do: tz_offset * 15 * 60
+  defp calc_utc_offset(%{std_offset: std_offset}, tz_offset) do
+    tz_offset_seconds = utc_offset_to_seconds(tz_offset)
+
+    if tz_offset < 0 do
+      tz_offset_seconds - std_offset
+    else
+      tz_offset_seconds + std_offset
+    end
```

Comment:
```suggestion
    # QMI reports the overall offset. Elixir splits the offset into the amount
    # that goes to standard time (utc_offset) + the daylight savings offset
    # from standard time (std_offset).
    utc_offset_to_seconds(tz_offset) - std_offset
```

Daylight savings time math is the same on both sides of the Earth.

Reply:
Arizona people struggles :smile: 

I updated.